### ISSUE TITLE
Layer System

### ISSUE THREAD

**Issue by obscurelyme** [obscurelyme/Semi-Pro-Cpp#3]
Currently (as of this writing 10/3/2021), the game engine only supports "layering" if the programmer is smart enough to know the exact order of how things are to be drawn to the screen. This works for small levels, but as the engine and games grow there are going to be more and more elements that will be considerably more difficult to keep track of. It would be helpful if each graphical element could be assigned a "layer", and from there items can be drawn in a FIFO manner. Alternatively I think it's best if the programmer had even more control and could specify the exact "z-index" property of certain elements sort of in a "position: absolute" kind of fashion. 

TLDR: The game engine needs a Layer System for all graphical elements. Think CSS [z-index ](https://developer.mozilla.org/en-US/docs/Web/CSS/z-index) capabilities.

**Comment by obscurelyme** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Layer System](https://trello.com/c/KCAm6fu7/4-layer-system)
